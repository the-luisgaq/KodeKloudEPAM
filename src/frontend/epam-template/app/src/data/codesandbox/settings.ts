import { Settings } from '@epam/uui';

/* eslint-disable */
export const settings_override: Settings | {} = "<UUI_SETTINGS_OVERRIDE>";
