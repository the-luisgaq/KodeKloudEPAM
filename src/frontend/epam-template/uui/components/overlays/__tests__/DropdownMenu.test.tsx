import React from 'react';
import {
    DropdownMenuBody, DropdownMenuButton, DropdownSubMenu, DropdownMenuSplitter, DropdownMenuHeader,
} from '../';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { settings } from '../../../settings';

const icon = settings.dropdownMenu.icons.dropdownIcon;

describe('DropdownMenu', () => {
    it('should be rendered DropdownMenuBody correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <DropdownMenuBody onClose={ () => {} } style={ { maxWidth: '250px' } }>
                <DropdownMenuButton caption="Menu Item in Submenu" />
                <DropdownMenuSplitter />
                <DropdownMenuHeader caption="An example of DropdownMenuHeader" />
                <DropdownSubMenu caption="One More SubMenu">
                    <DropdownMenuButton icon={ icon } iconPosition="right" caption="Menu Item with icon in right" />
                </DropdownSubMenu>
            </DropdownMenuBody>,
        );
        expect(tree).toMatchSnapshot();
    });
});
