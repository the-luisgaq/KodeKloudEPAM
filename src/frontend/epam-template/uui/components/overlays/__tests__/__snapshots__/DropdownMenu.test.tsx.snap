// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu should be rendered DropdownMenuBody correctly 1`] = `
<DocumentFragment>
  <div
    class="uui-dropdown-body -lock-focus root uui-dropdownMenu-body container"
    role="menu"
    tabindex="-1"
  >
    <div
      class="itemRoot"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="container"
      >
        Menu Item in Submenu
      </div>
    </div>
    <div
      class="splitterRoot"
    >
      <hr
        class="splitter"
      />
    </div>
    <div
      class="uui-dropdown-menu-header headerRoot"
    >
      <span
        class="header"
      >
        An example of DropdownMenuHeader
      </span>
    </div>
    <div
      class="submenuRootItem itemRoot"
      role="menuitem"
      tabindex="0"
    >
      <div
        class="container"
      >
        One More SubMenu
      </div>
      <div
        class="flexSpacer"
      />
      <button
        class="uui-button-box uui-enabled container uui-icon_button uui-color-neutral root icon iconAfter"
        tabindex="-1"
        type="button"
      >
        <div
          class="container uui-icon uui-enabled"
        >
          <svg
            class=""
          />
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
